### ISSUE TITLE
Merging Issues with Readme.md to derived repo

### ISSUE THREAD

**Issue by acrobat888** [jenstroeger/python-package-template#30]
Noticed when merging changes from this template repo to another repo the Readme.md file is "confusing."  I think there should be a Template_Readme.md file where all the relative template readme information is stored.

This allows me to create a new repo from this template and keep the _**derived**_ repo's Readme file separate, but also allows me to merge changes from this template easily without conflicts.

**Comment by jenstroeger** [comment]
@acrobat888 thank you, makes a lot of sense.

So, basically we want to rename the current README.md to TEMPLATE_README.md (or some such) and keep the new README.md empty for use by the actual forked repos? Can we have Github show the TEMPLATE_README.md on the repo’s landing page though? Hmm 🤔

**Comment by jenstroeger** [comment]
So, it looks like a `.github/README.md` takes precedence over the existing `README.md` file.

Considering other [community files like issue and PR templates](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions) (see also issue https://github.com/jenstroeger/python-package-template/issues/13) that are stored in the `.github/` folder, and taking issue https://github.com/jenstroeger/python-package-template/issues/31 into account as well, how about we

 - keep the existing [`README.md`](https://github.com/jenstroeger/python-package-template/blob/master/README.md) instead of renaming it to `TEMPLATE_README.md` (which Github wouldn’t pick up); and
 - keep the existing [`CHANGELOG.md`](https://github.com/jenstroeger/python-package-template/blob/master/CHANGELOG.md) instead of renaming it to `TEMPLATE_CHANGELOG.md`; and
 - document that users who build on this repository need to set up and use a `.github/README.md` and `.github/CHANGELOG.md` (alongside other, optional community files).

That would allow us to maintain the README and CHANGELOG independently from the user’s files, and a user can continue to merge updates to this repository into her own without conflicts.

@acrobat888, in that light I’d probably like to close issue https://github.com/jenstroeger/python-package-template/issues/31 as a duplicate of this one?

**Comment by behnazh** [comment]
Since `.github/README.md` takes precedence over `README.md`, it makes sense for this repo to use `README.md` then. And same goes with the `CHANGELOG.md`. 

While templates are useful for community files, I'm not sure they are necessary for `README.md` and `CHANGELOG.md`.

**Comment by acrobat888** [comment]
@jenstroeger I was thinking it would be better to move the `README.md` file into the _**docs**_ folder and leave a lite README file at the root with a link to the docs folder, or instructions to create a README.md at the root and **_link_** the original README.md which is in the docs folder.

I would be concerned that the `.github/README.md` would be missed and as it's always takes precedence.

**Comment by jenstroeger** [comment]
Turns out Github’s search order to find a README to display on the repos’s main/landing page is:

 - get the `.github/README.md`; if that doesn’t exist then
 - get the `README.md`; if that doesn’t exist then
 - get the `docs/README.md`; if that doesn’t exist then
 - display an alert box informing the user that no README file was found.

 So, we could move the current README (and CHANGELOG) into the `docs/` folder and ask users who fork this repo to add a README (and CHANGELOG) at the root of the repo — which would supersede the ones in the `docs/` folder.

I have one concern with this approach: the `docs/` folder is also the one where the [Python package documentation is generated](https://github.com/jenstroeger/python-package-template#documentation-1), but maybe that’s ok…

**Comment by jenstroeger** [comment]
Hmph. So. Turns out that if we set the git configuration to [`core.symlink = true`](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coresymlinks) then we’re able to check in symbolic links. That means we can rename `README.md` to `TEMPLATE_README.md` and create a symlink:

![symlink](https://user-images.githubusercontent.com/12053937/140861028-66ed7854-ec2d-4314-9276-cd1aa6dc6232.jpg)

That way Github picks up the `TEMPLATE_README.md` as the normal README and displays it, and users who fork the repo can then replace the symlink with their actual `README.md`. I’m just not sure how a merge would work if the symlink was replaced with an actual file (I can’t fork my own repo) — but we can find out? 😉

Shall I open a PR for this?
